File: arlo-admin-server/internal/modules/job/model/job.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// SysJob 定时任务
type SysJob struct {
	ID         uint64         `gorm:"primaryKey;autoIncrement" json:"id"`
	Name       string         `gorm:"size:64;not null" json:"name"`
	Handler    string         `gorm:"size:64;not null;index" json:"handler"`
	Cron       string         `gorm:"size:64;not null" json:"cron"`
	Params     string         `gorm:"size:512;not null;default:''" json:"params"`
	Status     int8           `gorm:"not null;default:1;index" json:"status"` // 0暂停 1启用
	Remark     string         `gorm:"size:255;not null;default:''" json:"remark"`
	LastRunAt  *time.Time     `json:"lastRunAt"`
	LastStatus *int8          `json:"lastStatus"` // 0失败 1成功
	CreatedAt  time.Time      `json:"createdAt"`
	UpdatedAt  time.Time      `json:"updatedAt"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`
}

func (SysJob) TableName() string { return "sys_job" }

// SysJobLog 执行日志
type SysJobLog struct {
	ID          uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	JobID       uint64    `gorm:"not null;index" json:"jobId"`
	JobName     string    `gorm:"size:64;not null;default:''" json:"jobName"`
	Handler     string    `gorm:"size:64;not null;default:''" json:"handler"`
	TriggerType int8      `gorm:"not null;default:0" json:"triggerType"` // 0调度 1手动
	Status      int8      `gorm:"not null;default:1" json:"status"`      // 0失败 1成功
	Result      string    `gorm:"type:text" json:"result"`
	ErrorMsg    string    `gorm:"size:1000;not null;default:''" json:"errorMsg"`
	DurationMs  int       `gorm:"not null;default:0" json:"durationMs"`
	CreatedAt   time.Time `json:"createdAt"`
}

func (SysJobLog) TableName() string { return "sys_job_log" }

File: arlo-admin-web/src/api/modules/job.ts
import request from '../request'

export interface JobItem {
  id: number
  name: string
  handler: string
  cron: string
  params: string
  status: number
  remark: string
  lastRunAt: string
  lastStatus: number | null
  nextRunAt: string
  createdAt: string
  updatedAt: string
}

export interface JobHandlerItem {
  code: string
  name: string
  description: string
}

export interface JobLogItem {
  id: number
  jobId: number
  jobName: string
  handler: string
  triggerType: number
  status: number
  result: string
  errorMsg: string
  durationMs: number
  createdAt: string
}

export function getJobList(params: {
  page: number
  pageSize: number
  name?: string
  handler?: string
  status?: number
}) {
  return request.get<{ list: JobItem[]; total: number; page: number; pageSize: number }>(
    '/v1/monitor/job/list',
    params,
  )
}

export function getJobDetail(id: number) {
  return request.get<JobItem>(`/v1/monitor/job/${id}`)
}

export function getJobHandlers() {
  return request.get<JobHandlerItem[]>('/v1/monitor/job/handlers')
}

export function createJob(data: {
  name: string
  handler: string
  cron: string
  params?: string
  status?: number
  remark?: string
}) {
  return request.post<JobItem>('/v1/monitor/job', data)
}

export function updateJob(id: number, data: {
  name: string
  cron: string
  params?: string
  remark?: string
}) {
  return request.put(`/v1/monitor/job/${id}`, data)
}

export function updateJobStatus(id: number, status: number) {
  return request.put(`/v1/monitor/job/${id}/status`, { status })
}

export function deleteJob(id: number) {
  return request.delete(`/v1/monitor/job/${id}`)
}

export function runJob(id: number) {
  return request.post<{ ok: boolean; msg: string }>(`/v1/monitor/job/${id}/run`)
}

export function getJobLogList(params: {
  page: number
  pageSize: number
  jobId?: number
  status?: number
  triggerType?: number
}) {
  return request.get<{ list: JobLogItem[]; total: number; page: number; pageSize: number }>(
    '/v1/monitor/job/log/list',
    params,
  )
}

export function getJobLogDetail(logId: number) {
  return request.get<JobLogItem>(`/v1/monitor/job/log/${logId}`)
}
